### PR TITLE
Add support for gcloudMode during Deploy

### PR DESCRIPTION
**Pull request by loosebazooka**
Allow deployments to use `alpha` or `beta` on `gcloud <mode> app deploy`

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/appengine/operations/Deployment.java [hunk 1]
```diff
@@ -25,15 +25,21 @@
...
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nullable;
 
 /** Deploy staged application and project configuration. */
 public class Deployment {
 
+  // optional gcloud modes
+  static final List<String> GCLOUD_MODES = ImmutableList.of("alpha", "beta");
```
Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/operations/Deployment.java]: ImmutableList<String> for clarity

File: src/main/java/com/google/cloud/tools/appengine/operations/Deployment.java [hunk 2]
```diff
@@ -146,4 +156,20 @@ void deployConfig(String filename, DeployProjectConfigurationConfiguration confi
...
+  @Nullable
+  String processMode(@Nullable String modeFromConfig) throws AppEngineException {
+    if (modeFromConfig == null) {
+      return null;
+    }
+    String trimmedModeFromConfig = modeFromConfig.trim();
+    Optional<String> matchingMode =
+        GCLOUD_MODES
+            .stream()
+            .filter(mode -> mode.equalsIgnoreCase(trimmedModeFromConfig))
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/appengine/operations/Deployment.java]: nit: `.filter(trimmedModeFromConfig::equalsIgnoreCase)` is more compact

I wouldn't care about supporting upper/mixed cases though, but this doesn't hurt.

File: src/main/java/com/google/cloud/tools/appengine/operations/Deployment.java (hunk 1 above)
Comment (chanseokoh) [reply src/main/java/com/google/cloud/tools/appengine/operations/Deployment.java]: nit: looks like this can be `private`.

**Comment by chanseokoh** [comment]
> Did a need for this surface somewhere?

https://github.com/GoogleCloudPlatform/app-maven-plugin/issues/374

**Comment by loosebazooka** [comment]
@elharo internal teams requesting it.

**Pull request merged**